account_financial_report/report/templates/trial_balance.xml

```diff
@@ -782,220 +753,238 @@
+                                  ('date', '&lt;=', account.report_id.date_to.strftime('%Y-%m-%d')),
+                                  ('period_balance', '&lt;&gt;', 0)]"
                     />
+                    <span>
+                        <a
+                            t-att-data-domain="domain"
+                            t-att-data-res-model="'account.move.line'"
+                            class="o_account_financial_reports_web_action_monetary_multi"
+                            t-att-style="style"
+                        >
+                            <t
+                                t-att-style="style"
+                                t-raw="account.period_balance"
+                                t-options="{'widget': 'monetary', 'display_currency': res_company.currency_id}"
+                            />
+                        </a>
+                    </span>
                 </div>
                 <!--## Ending balance-->
-                <div class="act_as_cell amount" style="width: 9%;">
-                    <span
-                        t-esc="total_amount[account_id]['ending_balance']"
-                        t-options="{'widget': 'monetary', 'display_currency': res_company.currency_id}"
+                <div class="act_as_cell amount" style="width: 9.64%;">
+                    <t
+                        t-set="domain"
+                        t-value="[('account_id', '=', account.account_id.id)]"
                     />
+                    <span>
+                        <a
+                            t-att-data-domain="domain"
+                            t-att-data-res-model="'account.move.line'"
+                            class="o_account_financial_reports_web_action_monetary_multi"
+                            t-att-style="style"
+                        >
+                            <t
+                                t-att-style="style"
+                                t-raw="account.final_balance"
+                                t-options="{'widget': 'monetary', 'display_currency': res_company.currency_id}"
+                            />
+                        </a>
+                    </span>
                 </div>
                 <t t-if="foreign_currency">
-                    <!--## currency_name-->
-                    <div class="act_as_cell" style="width: 4%;" />
-                    <!--## amount_total_due_currency-->
-                    <div class="act_as_cell" style="width: 9%;" />
-                    <!--## amount_residual_currency-->
-                    <div class="act_as_cell" style="width: 9%;" />
+                    <t t-if="account.currency_id.id">
+                        <!--## currency_name-->
+                        <div class="act_as_cell" style="width: 4.43%;">
+                            <t
+                                t-set="domain"
+                                t-value="[('account_id', '=', account.account_id.id)]"
```

Comment:
These domains should take into account the state, based on the value of 'only_posted_moves'.